File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.c
int maxDepth(char* s) {
    int count =0,max=0;
    for(int i=0;s[i]!='\0';i++){
        char ch=s[i];
        if(ch=='(') count++;
        else if(ch==')') count--;
        max=(count>max)?count:max;
    }
    return max;
}

File: 1614-maximum-nesting-depth-of-the-parentheses/1614-maximum-nesting-depth-of-the-parentheses.java
class Solution {
    public int maxDepth(String s) {
        int count=0,max=0;
        // for(int i=0;i<s.length();i++){
        //     if(s.charAt(i)=='('){
        //         count++;
        //         max=Math.max(max,count);
        //     }else if(s.charAt(i)==')'){
        //         count--;
        //     }
        // }
        for (char c : s.toCharArray()) {
            if (c == '(') {
                count++;
                max = Math.max(max, count);
            } else if (c == ')') {
                count--;
            }
        }
        return max;
    }
}